### DISCUSSION TITLE
How to block all sub Domains?

### DISCUSSION THREAD

**Discussion by LeroxHD** [RaspAP/raspap-webgui#797] [Q&A] [answered]
Hello,
I have added ,,0.0.0.0 facebook.com" to the Custom Blocklist in RaspAP, but now the Raspberry Pi only blocks the Domain facebook.com and not for example ads.facebook.com.
When I add all sub Domains manually, than it works. But Facebook alone has so many subdomains that it takes me years to add them all.

How do I make it that all subdomains are automatically blocked?

**Comment by billz** [comment, marked as answer]
It's possible to do this, but not with a hosts file, which the custom blocklist modifies.

If you want to block entire domains including subdomains, eg. `*.facebook.com` do the following:

1. edit RaspAP's adblock config `sudo nano /etc/dnsmasq.d/090_adblock.conf`
2. specify your own domain list by adding `conf-file=/etc/raspap/adblock/my-domains.txt`
3. edit this file `sudo nano /etc/raspap/adblock/my-domains.txt`
4. add the line `address=/facebook.com/0.0.0.0`
5. save, exit and restart Ad blocking
